libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/MapboxOffboardRouter.kt

```diff
@@ -14,42 +14,69 @@ import retrofit2.Callback
 import retrofit2.Response
 
 @MapboxNavigationModule(MapboxNavigationModuleType.OffboardRouter, skipConfiguration = true)
-class MapboxOffboardRouter(
-    private val context: Context,
-    private val mapboxToken: String
-) : Router {
+class MapboxOffboardRouter : Router {
+
+    companion object {
+        const val ERROR_FETCHING_ROUTE = "Error fetching route"
+    }
 
+    private val context: Context
+    private val mapboxToken: String
+    private val routeBuilderProvider: () -> NavigationRoute.Builder
     private var navigationRoute: NavigationRoute? = null
 
+    constructor(
+        context: Context,
+        mapboxToken: String
+    ) : this(
+        context,
+        mapboxToken,
+        {
+            NavigationRoute
+                .builder(context)
+                .accessToken(mapboxToken)
+        }
+    )
+
+    internal constructor(
+        context: Context,
+        mapboxToken: String,
+        routeBuilderProvider: () -> NavigationRoute.Builder
+    ) {
+        this.context = context
+        this.mapboxToken = mapboxToken
+        this.routeBuilderProvider = routeBuilderProvider
+    }
+
     override fun getRoute(
```

Comment:
It's pretty important to test the integration with the server side API:

- Sending the correct messages to the API.
- Parsing the API responses correctly.
- Implementing the authentication mechanisms properly.
- Handling API errors properly.

We need to simulate different server side responses and perform assertions over the client side request. To test if the API client is working as expected, we need to use test doubles and perform assertions over the information sent through the network.

To do so, we can use a ["MockWebServer"](https://github.com/square/okhttp/tree/master/mockwebserver). This library will work like a mock, but instead of replacing production code inside the API client, it will replace the production code implemented in server side.

cc @Lebedinsky